File: Sem3/Advanced Methods of Programming/MyInterpreter - CSharp Console Application/MyInterpreter CSharp/domain/expression/VariableExpression.cs
using MyInterpreter_CSharp.domain.adt;

namespace MyInterpreter_CSharp.domain.expression
{
    public class VariableExpression : Expression
    {
        private string _id;

        public VariableExpression(string id)
        {
            this._id = id;
        }

        public override int Eval(IMyDictionary<string, int> symbolTable)
        {
            try
            {
                return symbolTable.Lookup(_id);
            }
            catch (AdtException exception)
            {
                throw new ExpressionException(exception.Message + " " + _id);
            }
        }

        public override string ToString()
        {
            return _id;
        }
    }
}

File: Sem3/Advanced Methods of Programming/MyInterpreter - JavaFX - Final Project/project/src/domain/expression/VariableExpression.java
package domain.expression;


import domain.adt.AdtException;
import domain.adt.MyDictionaryInterface;


public class VariableExpression extends Expression
{
    private String id;


    public VariableExpression(String id)
    {
        this.id = id;
    }


    @Override
    public int evaluate(MyDictionaryInterface<String, Integer> symbolTable,
                        MyDictionaryInterface<Integer, Integer> heapTable) throws ExpressionException
    {
        try
        {
            return symbolTable.lookup(id);
        }
        catch (AdtException exception)
        {
            throw new ExpressionException(exception.getMessage() + " Variable expression.");
        }
    }


    @Override
    public String toString()
    {
        return id;
    }
}